workers/data_refinery_workers/processors/smasher.py

```diff
@@ -130,6 +131,18 @@ def _smash(job_context: Dict) -> Dict:
             experiments[experiment.accession_code] = experiment.to_metadata_dict()
         metadata['experiments'] = experiments
 
+        # Metadata to TSV
+        with open(smash_path + 'metadata.tsv', 'w') as output_file:
+            sample_ids = [k for k in metadata['samples'].keys()]
+            keys = [k for k in metadata['samples'][sample_ids[0]].keys()] + ['sample_id']
+            dw = csv.DictWriter(output_file, keys, delimiter='\t')
+            dw.writeheader()
+            for key, value in metadata['samples'].items():
+                value['sample_id'] = key
+                dw.writerow(value)
+
+        metadata['files'] = os.listdir(smash_path)
```

Comment:
After #351, this looks like it could be potentially problematic at some point. Looks like you're using `str(job_context["dataset"].pk)` to make this directory specific to this dataset. However couldn't there be eventually be a scenario where a job which puts one or more files into this directory, gets either interrupted or fails for a reason which doesn't happen a second time, then the foreman restarts it causing a non-empty directory to be used. While it seems like we probably won't put extra files in there cause the names would collide, potentially we could append to an existing file? It maybe wouldn't ever be an issue, but I think we eliminate the potential for it to happen if up where we do:
```
# Prepare the output directory
smash_path = "/home/user/data_store/smashed/" + str(job_context["dataset"].pk) + "/"
os.makedirs(smash_path, exist_ok=True)
```
We first make sure the directory doesn't exist and remove it and then recreate it if it does.